controlpanel/api/models/tool.py

```diff
@@ -204,3 +204,68 @@ def restart(self, id_token):
         """
 
         cluster.ToolDeployment(self.user, self.tool).restart(id_token)
+
+
+class HomeDirectoryManager:
+    """
+    Emulates a Django model manager.
+    """
+
+    def create(self, *args, **kwargs):
+        home_directory = HomeDirectory(*args, **kwargs)
+        home_directory.save()
+        return home_directory 
+
+    def filter(self, **kwargs):
+        return []
+
+
+class HomeDirectory:
```

Comment:
I don't think this should be among the models as it's more of nice way to interact with the cluster (and possibly not in the `api.models.tool` but this is a WIP PR so I know any of this is subject to change ☺️ )

I'd move this class in the `cluster` module where the other classes we use to interact with 3rd party systems are.

The views can still use this, don't think we need to have the model when in this case there is nothing really persisted in the DB (as the state is in k8s cluster).

Reply:
Again, just following the coding conventions found in the code base. Again, I did wonder, and agree cluster is probably a better place to put it.